src/fingerprint.ts

```diff
@@ -10,7 +10,7 @@ import type {ScriptReferenceString} from './script.js';
  * All meaningful inputs of a script. Used for determining if a script is fresh,
  * and as the key for storing cached output.
  */
-export interface Fingerprint {
+export interface FingerprintData {
```

Comment:
It occurs to me that the order of fields matters in FingerprintData, since they're compared by JSON serialization. I think that should be documented here.

It may be worthwhile to more tightly control construction. For example by adding a brand on the type, as syntactic vinegar to make a speed bump when constructing a FingerprintData.

Reply:
Moved creation into the fingerprint class itself and added a brand, so that it's hard to create a fingerprint except through that class.